docs/standard/io/file-path-formats.md

```diff
@@ -0,0 +1,171 @@
+
+If the path starts with something other than a separator, the current drive and current directory are applied. For example, if the path is `filecompare` and the current directory is `C:\utilities\`, the result is `C:\utilities\filecompare\`.
+
+> [!IMPORTANT]
+> Relative paths are dangerous in multithreaded applications (that is, most applications) because the current directory is a per-process setting. Any thread can change the current directory at any time. 
+
+### Canonicalizing separators
+
+All forward slashes (`/`) are converted into the standard Windows separator, the back slash (`\`). If they are present, series of slashes that follow the first two slashes are collapsed into a single slash.
+
+> [!IMPORTANT]
+> It is only when identifying paths for normalization purposes that the direction of the slash does not matter. Forward slashes are not supported in Windows outside of this normalization step. This is critically important if normalization is skipped.
+
+### Evaluating relative components
+
+As the path is processed, any components or segments that are composed of a single or a double period (`.` or `..`) are evaluated: 
+
+- For a single period, the current segment is removed, since it refers to the current directory).
+
+- For a double period, the current segment and the parent segment are removed, since the double period refers to the parent directory.
+
+   Parent directories are only removed if they aren't past the root of the path. The root of the path depends on the type of path. It is the drive (`C:\`) for DOS paths, the server/share for UNCs (`\\Server\Share`), and the device path prefix for device paths (`\\?\` or `\\.\`).
+
+### Trimming characters
+
+Along with the runs of separators and relative segments removed earlier, some additional characters are removed during normalization:
+
+- If a segment ends in a single period, that period is removed. (A segment of a single or double period is normalized in the previous step. A segment of three or more periods is not normalized and is actually a valid file/directory name.
+
+- If the path doesn't end in a separator, all trailing periods and spaces (U+0020) are removed. If the last segment is simply a single or double period, it falls under the relative components rule above. 
+
+   This rule leads to the possibly surprising ability to create a directory with a trailing space. You simply need to add a trailing separator to do so.
```

Comment:
Do you mean "surprising inability" instead of "surprising ability"?

Reply:
@BillWagner, no, surprising ability. The space is included with, for example, "directoryName \" But "surprising ability" is not a good phrase, as your comment suggests. I'll change it to "This rule mans that you can create a directory name with a trailing space by adding a trailing separator after the space. Trailing spaces can make it difficult to access a directory."

Reply:
Please make sure to **strongly** discourage adding spaces or periods at the end of directory/filenames. Most applications fall over, including anything based on .NET. I fixed a number of issues around this in 2.1, particularly around enumeration, but that shouldn't change the guidance. Test for it, sure, but don't introduce them. :)